src/productcatalogservice/server.go

```diff
@@ -282,7 +328,54 @@ func (p *productCatalog) GetProduct(ctx context.Context, req *pb.GetProductReque
 	if found == nil {
 		return nil, status.Errorf(codes.NotFound, "no product with ID %s", req.Id)
 	}
-	return found, nil
+
+	conn, err := grpc.DialContext(ctx, p.addr, grpc.WithInsecure(), grpc.WithTimeout(time.Second*3), grpc.WithStatsHandler(&ocgrpc.ClientHandler{}))
+	if err != nil {
+		return nil, err
+	}
+
+	client := discountservice.NewDiscountServiceClient(conn)
+	d, err := client.GetProductDiscount(ctx, convertProductDiscount(found))
+	if err != nil {
+		return nil, err
+	}
+	return convertProduct(d), nil
+}
+
+func convertProductDiscount(product *pb.Product) *discountservice.Product{
```

Comment:
We should probably rename `convertProductDiscount` and `convertProduct` to make clearer what its purpose is :)